### PR TITLE
Enable PKCE by default for Keycloak JS

### LINKED ISSUE #26411: Enable PKCE by default in Keycloak JS
As part of providing more secure defaults, Keycloak JS should be enabling Proof Key Code Exchange ([PKCE](https://datatracker.ietf.org/doc/html/rfc7636)) by default, with the ability to opt-out if needed.

### PR DESCRIPTION
**Pull request by jonkoops**
Closes #26411

### CODE REVIEW

**Review by agagancarczyk — DISMISSED** [review]
LGTM!

**Review by edewit — DISMISSED** [review]

**Comment by jonkoops** [comment]
@andymunro could I ask you to vet the changes I made to the documentation here?

**Review by edewit — DISMISSED** [review]

**Review by stianst — DISMISSED** [review]

**Review by agagancarczyk — DISMISSED** [review]
LGTM!

**Review by hmlnarik — CHANGES_REQUESTED** [review]
It seems that the default is never set, otherwise looks good to me, thanks!

File: js/libs/keycloak-js/src/keycloak.js [hunk 1]
```diff
@@ -134,9 +134,11 @@ function Keycloak (config) {
 
             if (initOptions.pkceMethod) {
                 if (initOptions.pkceMethod !== "S256") {
-                    throw 'Invalid value for pkceMethod';
+                    throw new TypeError(`Invalid value for 'pkceMethod', expected 'S256' but got '${initOptions.pkceMethod}'.`);
                 }
                 kc.pkceMethod = initOptions.pkceMethod;
+            } else {
+                kc.pkceMethod = "S256";
```
Comment (jonkoops) [inline js/libs/keycloak-js/src/keycloak.js:141]: Had forgotten to add the default, can I ask for another round of reviews?

**Review by hmlnarik — APPROVED** [review]
Fixed with latest commit, approving. Thanks!

**Pull request merged**